src/Compilers/Core/Portable/MetadataReference/MetadataReference.cs

```diff
@@ -257,9 +257,15 @@ public MetadataReference WithProperties(MetadataReferenceProperties properties)
         /// Reusing <see cref="AssemblyMetadata"/> object allows for sharing data accross these references.
         /// </para>
         /// </remarks>
+        [Obsolete("The API is obsolete, use CreateFromFile(assembly.Location) instead", error: true)]
         public static MetadataReference CreateFromAssembly(Assembly assembly)
         {
-            return CreateFromAssembly(assembly, default(MetadataReferenceProperties));
+            throw new NotImplementedException();
+        }
```

Comment:
We shouldn't throw the exception here.  The point is to keep binary compat so that people who are already calling this.  Instead, just have it call `CreateFromAssemblyInternal` directly, so that existing code continues to work where it works now, but fails with `PlatformNotSupported` exception if ported to new platforms.